### PR TITLE
[NXP K32W] Reduce heap usage by reducing event queue length

### PR DESCRIPTION
**Pull request by tcarmelveilleux**
### Problem
Issue #9261

- K32W platform ran out of BSS last week, and had to reduce heap size
- Current usage is default max 100 events in queue. PR #17959 already reduced
  the size of each event and saved approx 600 bytes. This PR should reduce it
  further.
- This change reduces the maximum amount of heap that could be used by events, to save more heap headroom

#### Change overview
- This PR moves max queue size from 100 to 75, which is 25% reduction
  and much larger than some platforms which use 25 (which seems low to me)

- Other Thread platforms like QPG and EFR32 run on 25, so not worried here

#### Testing
- Sanity check on K32W via @doru91

### CODE REVIEW

File: src/platform/nxp/k32w/k32w0/CHIPDevicePlatformConfig.h
Comment (tcarmelveilleux) [inline src/platform/nxp/k32w/k32w0/CHIPDevicePlatformConfig.h:77]: Wait for @doru91 test success confirmation before merge

**Review by bzbarsky-apple — APPROVED** [review]

**Review by msandstedt — APPROVED** [review]

**Review by doru91 — APPROVED** [review]

**Comment by doru91** [comment]
Tested with success a commissioning flow. Please note that this will help with heap usage reduction, as the event queue is dynamically allocated.

**Pull request merged**
